Testsuite for Solaris when building 32 bit breaks

When building the latest version on Solaris 10 with 32 bit the test suite does not compile:
```
gmake  test_common test_meta_data test_utils_avltree test_utils_heap test_utils_time test_utils_subst
gmake[4]: Entering directory '/home/dam/work/collectd/src/daemon'
  CC       common_test.o
In file included from collectd.h:31:0,
                 from common.h:31,
                 from common_test.c:28:
../../src/config.h:1622:0: error: "_FILE_OFFSET_BITS" redefined [-Werror]
 #define _FILE_OFFSET_BITS 64
 ^
In file included from /usr/include/inttypes.h:20:0,
                 from ../../src/testing.h:27,
                 from common_test.c:27:
/opt/csw/lib/gcc/i386-pc-solaris2.10/4.9.2/include-fixed/sys/feature_tests.h:196:0: note: this is the location of the previous definition
 #define _FILE_OFFSET_BITS 32
 ^
cc1: all warnings being treated as errors
Makefile:986: recipe for target 'common_test.o' failed
```

This has probably to do with largefiles as described in lf64(5) being different in the standard GCC headers and the determined width.

The issue reappeared in the latest master:
```
  CCLD     write_http.la
gmake  test_format_json test_utils_latency test_utils_vl_lookup test_utils_mount test_plugin_ceph
gmake[4]: Entering directory '/home/dam/work/collectd/src'
  CC       utils_format_json_test.o
In file included from ./daemon/collectd.h:31:0,
                 from utils_format_json_test.c:28:
./config.h:1671:0: error: "_FILE_OFFSET_BITS" redefined [-Werror]
 #define _FILE_OFFSET_BITS 64
 ^
In file included from /usr/include/inttypes.h:20:0,
                 from testing.h:30,
                 from utils_format_json_test.c:27:
/opt/csw/lib/gcc/sparc-sun-solaris2.10/5.2.0/include-fixed/sys/feature_tests.h:196:0: note: this is the location of the previous definition
 #define _FILE_OFFSET_BITS 32
 ^
utils_format_json_test.c:45:3: error: conflicting types for 'label_t'
 } label_t;
   ^
In file included from /usr/include/sys/types.h:23:0,
                 from ./daemon/collectd.h:36,
                 from utils_format_json_test.c:28:
/usr/include/sys/machtypes.h:33:42: note: previous declaration of 'label_t' was here
 typedef struct _label_t { long val[2]; } label_t;
                                          ^
cc1: all warnings being treated as errors
Makefile:4130: recipe for target 'utils_format_json_test.o' failed
gmake[4]: *** [utils_format_json_test.o] Error 1
```

Hmm this breaks python, because it knows better:

```
In file included from ./daemon/collectd.h:31:0,
                 from python.c:32:
./config.h:1683:0: error: "_REENTRANT" redefined [-Werror]
 #define _REENTRANT /**/
 ^
In file included from /usr/include/python2.6/Python.h:8:0,
                 from python.c:27:
/usr/include/python2.6/pyconfig.h:999:0: note: this is the location of the previous definition
 #define _REENTRANT 1
 ^
```
https://bugs.python.org/issue1045893

Is it too early in the morning to start drinking yet?

--- a/src/utils_format_json_test.c
+++ b/src/utils_format_json_test.c
@@ -24,6 +24,16 @@
  *   Florian octo Forster <octo at collectd.org>
  */
 
+/* Workaround for Solaris 10 defining label_t
+ * Issue #1301
+ */
+#if KERNEL_SOLARIS
+# ifndef _POSIX_C_SOURCE
+#  define _POSIX_C_SOURCE 200112L
+# endif
+# undef __EXTENSIONS__
+#endif
+
 #include "collectd.h"
 
 #include "testing.h"

